test/ReactWrapper-spec.jsx

```diff
@@ -2051,6 +2051,12 @@ describeWithDOM('mount', () => {
       expect(parents.at(1).hasClass('bar')).to.equal(true);
       expect(parents.at(2).hasClass('bax')).to.equal(true);
     });
+
+    it('should work with components in the tree', () => {
+      const Foo = () => (<div className="bar" />);
+      const wrapper = mount(<Foo />);
+      expect(wrapper.find('.bar').parent()).to.equal(wrapper.find('Foo'));
```

Comment:
Done. Shallow wrapper returns empty wrapper in this case.

Also, I have updated test for mounted wrapper. Now it expects that it has`.root` as a parent. It is not possible to do the same for shallow because it doesn't render components deeply.